ui/src/utils/oauth.js

```diff
@@ -78,29 +78,80 @@ function getEndpoint(provider) {
 }
 
 export function openPopup(provider) {
+  console.log('openPopup', provider);
   return window.open(getEndpoint(provider), provider, `${settings},${getPopupDimensions(provider)}`);
 }
 
+/**
+ * @typedef {object} OauthError
+ *  @property {string} message
+ */
+
+/**
+ * @callback checkForToken~tokenResolve
+ * @param {string} token
+ */
+
+/**
+ * @callback checkForToken~tokenReject
+ * @param {OauthError} error
+ */
+
+/**
+ * @param {Window} popup
+ * @param {checkForToken~tokenResolve} resolve
+ * @param {checkForToken~tokenReject} reject
+ */
 function checkForToken(popup, resolve, reject) {
-  if (popup.closed) reject({ errors: 'Authentication was cancelled.' });
-  else {
+  console.log('checkForToken');
```

Comment:
Again should probably remove this.

Reply:
Ah damn 🙂 Maybe we should setup eslint rule for debug messages?

Reply:
Yeah agreed.